Lapka.Pets.Infrastructure/Services/PetQueryService.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Convey.Persistence.MongoDB;
+using Lapka.Pets.Application.Services;
+using Lapka.Pets.Core.Entities;
+using Lapka.Pets.Infrastructure.Documents;
+
+namespace Lapka.Pets.Infrastructure.Services
+{
+    public class PetQueryService : IPetQueryService
+    {
+        private readonly IMongoRepository<PetDocument, Guid> _repository;
+
+        public PetQueryService(IMongoRepository<PetDocument, Guid> mongoRepository)
+        {
+            _repository = mongoRepository;
+        }
+        
+        public async Task<Pet> GetByIdAsync(Guid id)
+        {
+            PetDocument petFromDb = await _repository.GetAsync(id);
+            if (petFromDb is null) return null;
```

Comment:
In this case throw exception that document not found, and map this for 404 not found